### DISCUSSION TITLE
前端dist包放到后端项目打成一个 jar 包运行时，页面刷新后，pinia的持久化的数据会丢失，是什么问题？

### DISCUSSION THREAD

**Discussion by lixian207** [vuejs/pinia#2442] [Help and Questions]
![image](https://github.com/vuejs/pinia/assets/20462414/ac64fc1c-0988-4871-9890-7f7b731348fd)
持久化代码：
![image](https://github.com/vuejs/pinia/assets/20462414/8bf1c5c5-c535-440d-a393-3de9085cf800)

页面未刷新前一切正常：
![image](https://github.com/vuejs/pinia/assets/20462414/6fb6f15b-4445-4c4a-9788-44c84d22f1c3)

页面刷新后，sessionstorage的数据丢失：
![image](https://github.com/vuejs/pinia/assets/20462414/d232a5af-24b8-4862-950a-446ab30cdd66)

这是什么原因造成的？
如果前端单独使用 nginx 部署，刷新后，pinia的持久化的数据不会丢失，是 pinia 依赖什么环境吗？

**Comment by kinds7** [comment]
考虑将路由方式改为hash，或者修改nginx配置
